### PR TITLE
DOCS: Align.py changes made 

### LINKED ISSUE #3365: Failed doc examples in align.py
## Expected behavior ##

Sub-issue of #760

Doc examples work.

## Actual behavior ##

This is after changing a `>>>` in the rotation matrix example to `...`

```
================================================== FAILURES ===================================================
_____________________________________ [doctest] MDAnalysis.analysis.align _____________________________________
069    >>> from MDAnalysis.analysis.rms import rmsd
070    >>> from MDAnalysis.tests.datafiles import PSF, DCD, PDB_small
071
072
073 In the simplest case, we can simply calculate the C-alpha RMSD between
074 two structures, using :func:`rmsd`::
075
076    >>> ref = mda.Universe(PDB_small)
077    >>> mobile = mda.Universe(PSF,DCD)
078    >>> rmsd(mobile.select_atoms('name CA').positions, ref.select_atoms('name CA').positions)
Expected:
    16.282308620224068
Got:
    28.20178579474479

/Users/lily/pydev/mdanalysis/package/MDAnalysis/analysis/align.py:78: DocTestFailure
_____________________________ [doctest] MDAnalysis.analysis.align.rotation_matrix _____________________________
238         RMSD between `a` and `b` before rotation
239     ``(R, rmsd)`` rmsd and rotation matrix *R*
240
241     Example
242     -------
243     `R` can be used as an argument for
244     :meth:`MDAnalysis.core.groups.AtomGroup.rotate` to generate a rotated
245     selection, e.g. ::
246
247     >>> R = rotation_matrix(A.select_atoms('backbone').positions,
UNEXPECTED EXCEPTION: NameError("name 'A' is not defined")
Traceback (most recent call last):
  File "/Users/lily/anaconda3/envs/mda-dev/lib/python3.7/doctest.py", line 1329, in __run
    compileflags, 1), test.globs)
  File "<doctest MDAnalysis.analysis.align.rotation_matrix[0]>", line 1, in <module>
NameError: name 'A' is not defined
/Users/lily/pydev/mdanalysis/package/MDAnalysis/analysis/align.py:247: UnexpectedException
=========================================== short test summary info ===========================================
FAILED analysis/align.py::MDAnalysis.analysis.align
FAILED analysis/align.py::MDAnalysis.analysis.align.rotation_matrix
```

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda
from MDAnalysis.tests.datafiles import PSF, DCD,  GRO, PDB, TPR, XTC, TRR,  PRMncdf, NCDF

u = mda.Universe(PSF, DCD)

....

```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) https://github.com/MDAnalysis/mdanalysis/commit/8f9f402fe65d6a90e88b60b6da735c130425f00e
- Which version of Python (`python -V`)? 3.7.3
- Which operating system? Mac Catalina

### PR DESCRIPTION
**Pull request by MohitKumar020291**
Fixes #3365

Changes made in this Pull Request: This PR is resolving suggestions by @orbeckst in #4023.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

----
:books: Documentation preview :books:: https://mdanalysis--4182.org.readthedocs.build/en/4182/

### CODE REVIEW

File: package/MDAnalysis/analysis/align.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: You need to use TPR and TRR here.

**Review by orbeckst — CHANGES_REQUESTED** [review]
# Does your solution work?

Have you checked that your solution works? Show a screen shot when you run your code in a Python session.

Have you run the doc tests? According to #3925 you can run doc tests with the commands
```python
cd package/doc/sphinx
make doctest
```
or alternatively follow https://github.com/MDAnalysis/mdanalysis/issues/760#issuecomment-882578843 and invoke via `pytest`
```
cd package/MDAnalysis
pytest . --disable-pytest-warnings --doctest-modules
```
Does the test pass now? Show evidence: Show that you _don't_ have the line
```
FAILED analysis/align.py::MDAnalysis.analysis.align.rotation_matrix
```
anymore.

# New author
- [x] Add yourself to the file AUTHORS.
- [x] Send an email to the developer list introducing yourself (and mention this PR); if you've already done so, add a link to the email in the list archive in the comments of this PR.

**Comment by MohitKumar020291** [comment]
@orbeckst I have been facing this problem related `__authors__`
https://i.imgur.com/arTENLD.png

**Comment by orbeckst** [comment]
The AUTHORS file is [package/AUTHORS](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/AUTHORS). 

The error that you encounter may be due to not having installed MDAnalysis. 

I suggest you follow the instructions in the user guide to create a development environment with conda https://userguide.mdanalysis.org/stable/contributing_code.html#creating-a-development-environment and then install a development version with `pip install -e ./package`.

If it still fails, describe step by step what you did. (Copy and paste code into the issue, that’s better than images; if only images are possible then you can directly drop them here, too)

**Comment by MohitKumar020291** [comment]
@orbeckst thanks for review. I tried multiple ways of creating universe but doctests failed. I am attaching the screenshots, please help to solve them.
https://i.imgur.com/xCEdQbE.png
https://i.imgur.com/XRbeVKl.png

**Comment by RMeli** [comment]
@MohitKumar020291 the second screenshot corresponds to the issue outlined by @orbeckst. In the first screenshot I think the issue is that you are using TPR and TRR within strings, while they should be passed directly.

**Comment by MohitKumar020291** [comment]
Thanks @RMeli for review. I tried passing TRR and TPR directly. But the results are not desirable. Screenshot attached.
https://i.imgur.com/U064e9k.png
**EDIT**: Please ignore, it is fixed now.

**Comment by MohitKumar020291** [comment]
@RMeli and @orbeckst. I have made some latest change. I imported align as to use `rotation_matrix`. Secondly, I skipped one line because it was causing wrong output, Attached screenshots:


-----------------------------------------------------------------
RELATED TO ABOVE IMAGE

--------------------------------------------------------------------------------------------------------------------------------------------
After doing these changes I am **not** getting Failed test(like `FAILED analysis\align.py::MDAnalysis.analysis.align.rotation_matrix
`), Screenshot attached(THIS IMAGE DOES NOT HAVE Failing `align.rotation_matrix` as of SECOND IMAGE ):



This solves one out of two problems of `align.py`.

File: package/MDAnalysis/analysis/align.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: Why do we have to skip the doc test here? We'd like to check that _everything_ works.

Doc test expects an output here so just add it: figure out what `A.atoms.rotate(R)` prints and then add it as a line after (and remove the +SKIP).

**Review by orbeckst — CHANGES_REQUESTED** [review]
Let's avoid skipping and make the doc test fully work (see inline comment).

Note I can quickly run the doc test with
```bash
cd package/MDAnalysis
pytest -v --disable-pytest-warnings --doctest-modules analysis/align.py
```
I suggest you remove the `# doctest: +SKIP`, then run the above command. It will immediately tell you all you need to know to fix the problem.

Please also add yourself to AUTHORS and introduce yourself on the mailing list (see my previous review comments) — we won't merge a new contribution without the authorship/email information.

**Comment by orbeckst** [comment]
> This solves one out of two problems of align.py.

Great that you identified another problem in the same file. It would be great if all tests in align.py were to pass — please, feel free to also fix this in the same PR here! It looks as if it's just a missing closing parenthesis.

```
=================================== FAILURES ===================================
_____________________ [doctest] MDAnalysis.analysis.align ______________________
083 mass (or geometry) first:
084
085    >>> rmsd(mobile.select_atoms('name CA').positions, ref.select_atoms('name CA').positions, center=True)
086    21.892591663632704
087
088 This has only done a translational superposition. If you want to also do a
089 rotational superposition use the superposition keyword. This will calculate a
090 minimized RMSD between the reference and mobile structure.
091
092    >>> rmsd(mobile.select_atoms('name CA').positions, ref.select_atoms('name CA').positions,
UNEXPECTED EXCEPTION: SyntaxError("'(' was never closed", ('<doctest MDAnalysis.analysis.align[8]>', 1, 5, "rmsd(mobile.select_atoms('name CA').positions, ref.select_atoms('name CA').positions,\n", 1, 0))
Traceback (most recent call last):
  File "~/anaconda3/envs/mda310dev/lib/python3.10/doctest.py", line 1348, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest MDAnalysis.analysis.align[8]>", line 1
    rmsd(mobile.select_atoms('name CA').positions, ref.select_atoms('name CA').positions,

SyntaxError: '(' was never closed
```

**Comment by orbeckst** [comment]
I had a quick look at align.py: there are a few more errors. The "missing parenthesis" is only the first one (and it's not really a missing parenthesis, just an incorrect line continuation).

If you want to, you can quickly work through all of them and make this a very nice PR:

1. First run the doc test
   ```python
   pytest -v --disable-pytest-warnings --doctest-modules package/MDAnalysis/analysis/align.py
   ```
   just on the file.
2. Then read the error message.
3. Fix the line.
4. Repeat with 1

At the end there's an example that uses the clustalw program: just skip all of it with
```python
   >>> seldict = align.fasta2select('sequences.aln')                                # doctest:+SKIP
   >>> alignment = align.AlignTraj(trj, ref, filename='rmsfit.dcd', select=seldict) # doctest:+SKIP
   >>> alignment.run()                                                              # doctest:+SKIP
```
as this example is difficult to run under all circumstances (because clustalw is not always installed) and we don't have all input files at hand.

**Comment by MohitKumar020291** [comment]
@orbeckst and @RMeli. I have again updated `align.py` and these changes helped to `Pass` both `failures` of `align.py`.
Screenshot attached:


--------------------------------------------------------------
Please review.

**Comment by MohitKumar020291** [comment]
> Please also add yourself to AUTHORS and introduce yourself on the mailing list (see my previous review comments) — we won't merge a new contribution without the authorship/email information.

I tried finding "how to add myself to changelog and author?", but did not found helpful results. So, I tried copying format in AUTHOR but I can't see 2023 in package/AUTHORS, so, I created my one(2023) which is obviously causing errors. Any solution to it?

File: package/MDAnalysis/analysis/align.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: This line produces output with a variable element. Use ellipsis as in https://stackoverflow.com/a/1024426 to also test for it.

File: package/CHANGELOG
Comment (orbeckst) [inline package/CHANGELOG:16]: I fixed the CHANGELOG for you. You needed to merge develop into your PR.

For doc test fixes and similar changes that are not user visible we typically only record you as a contributor with your GitHub handle @MohitKumar020291 and don't list the specific change.

You don't have to make any more changes in CHANGELOG.

**Review by orbeckst — CHANGES_REQUESTED** [review]
Please fix the one test that I mentioned in the comments.

The AUTHORS file can be found in the checked out sources as [`package/AUTHORS`](https://github.com/MohitKumar020291/mdanalysis/blob/3365/package/AUTHORS). When you are in the top level directory that you got from `git clone` you should be able to `cd packages` and then edit the AUTHORS file.

**Comment by orbeckst** [comment]
I am trying to understand why it's difficult for you to find the AUTHORS file. How do you edit the files that you are changing?

**Comment by MohitKumar020291** [comment]
> I am trying to understand why it's difficult for you to find the AUTHORS file. How do you edit the files that you are changing?

@orbeckst I can find the AUTHORS file but not 2023 it has only 2022. I was trying to change it through `edit`(it is a bad practice) but it is also not showing in my VS code(on both develop and 3365 branch).

Apart from that I have resolved the latest changes. 
Thanks for review

**Comment by RMeli** [comment]
@MohitKumar020291 you might have an older version of the repository and the errors you see are conflicts?

**Comment by MohitKumar020291** [comment]
@RMeli I have latest version even then I will check it again. Yes, they were conflicts.

**Comment by orbeckst** [comment]
THanks for introducing yourself on the developer list.

The current AUTHORS file is on your branch: I can see it in the web interface https://github.com/MohitKumar020291/mdanalysis/blob/3365/package/AUTHORS. Maybe you have to `git pull` your local branch in order to update it? If there are merge conflicts then you have to fix them.

EDIT: The AUTHORS addition is the only thing that's missing before we can approve the PR and squash-merge.

**Comment by MohitKumar020291** [comment]
@orbeckst I have successfully updated AUTHORS file. Thanks.

File: package/MDAnalysis/analysis/align.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: I don't think that's needed if you use the standard `...` ellipsis. Don't redefine the marker.
```suggestion
```

File: package/MDAnalysis/analysis/align.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: Although your solution worked, we can do this even simpler (with fewer distracting additional lines of code)
```suggestion
   >>> alignment.run()
   <MDAnalysis.analysis.align.AlignTraj object at ...>
```

**Review by orbeckst — CHANGES_REQUESTED** [review]
Thanks for adding yourself. I looked at your solution again. ALthough it works, it can be written in a more succinct manner. In order to keep the example code as simple as possible, I'd like you to change your code. I made suggestions that you should be able to simply accept and include. Thanks!

**Comment by MohitKumar020291** [comment]
@orbeckst changes done. Thanks.

**Review by orbeckst — APPROVED** [review]
Looking good! Thank you for your contribution.

I'll just wait until the CI comes back all green and then squash-merge.

**Review by orbeckst — CHANGES_REQUESTED** [review]
There's now an issue with actually building the documentation, see https://github.com/MDAnalysis/mdanalysis/actions/runs/5439978586/jobs/9892463278?pr=4182#step:5:3019 . There's something wrong with the formatting. Please investigate. The *build docs* check must also pass.

It is possible that you need to indent the code blocks under `::`.

**Comment by MohitKumar020291** [comment]
@orbeckst I tried finding issues multiple times with formatting but I did not get anything.

**Comment by orbeckst** [comment]
I'd recommend that you 

1. build the docs locally
2. check if it fails
3. fix file/try out things
4. goto 1 (until no errors)
5. commit & push
6. check CI

instead of having to wait for a while until the CI on GitHub runs.

To build the docs locally have a look at our [User Guide: Building docs](https://userguide.mdanalysis.org/dev/contributing_code.html#building-the-documentation). You will need to set up a development environment in which you can build docs. Follow [User Guide: Create a virtual environment]( https://userguide.mdanalysis.org/stable/contributing_code.html#create-virtual-environment) (I would use conda/mamba). Unfortunately, the docs are a bit outdated. https://github.com/MDAnalysis/UserGuide/issues/260 contains some instructions to be added to an updated User Guide version. In short, try in your conda environment (with python>=3.8):
```
# packages for MDAnalysis
conda install -c bioconda -c conda-forge \
            biopython chemfiles clustalw==2.1 codecov cython \
            griddataformats gsd hypothesis "joblib>=0.12" \
            matplotlib mmtf-python mock netcdf4 networkx \
            "numpy>=1.18.0" psutil pytest scikit-learn scipy \
            "seaborn>=0.7.0" "tidynamics>=1.0.0" \
            "tqdm>=4.43.0"

# documentation dependencies
conda install -c conda-forge sphinx pybtex pybtex-docutils \
	   sphinxcontrib-bibtex sphinx_rtd_theme sphinx-sitemap
python -m pip install msmb_theme==1.2.0
```

Then you should be able to build the docs as described, in short
```bash
cd package
python setup.py build_sphinx -E
```
Look at the output to see where files are generated. Read the output for errors.

**Comment by MohitKumar020291** [comment]
@orbeckst I think I have fixed the issue build docs, this is failing because of code coverage. Can you please rerun the checks?

**Review by orbeckst — APPROVED** [review]
Looks good! 🎉 

The codecov thing is an annoying spurious failure and not related to your PR.

**Comment by orbeckst** [comment]
Well done! Thank you for your contribution!

**Comment by MohitKumar020291** [comment]
@orbeckst and @RMeli, Thanks to both for reviews and great guidance.

**Pull request merged**
